tests/scattering1d/test_timefrequency_scattering.py

```diff
@@ -151,9 +151,111 @@ def test_frequency_scattering():
+    S = TimeFrequencyScatteringBase(
+        J=J, J_fr=J_fr, shape=shape, Q=Q, T=0, F=0, backend='numpy')
+    S.build()
+    S.create_filters()
+    assert not S.average
+    assert not S.average_fr
+    backend = kymatio.Scattering1D(
+        J=J, Q=Q, shape=shape, T=0, backend='numpy').backend
+
+    x = torch.zeros(shape)
+    x[shape[0]//2] = 1
+    x_shape = backend.shape(x)
+    batch_shape, signal_shape = x_shape[:-1], x_shape[-1:]
+    x = backend.reshape_input(x, signal_shape)
+    U_0_in = backend.pad(x, pad_left=S.pad_left, pad_right=S.pad_right)
+
+    filters = [S.phi_f, S.psi1_f, S.psi2_f]
+    jtfs_gen = joint_timefrequency_scattering(U_0_in, backend,
+        filters, S.oversampling, S.average=='local',
+        S.filters_fr, S.oversampling_fr, S.average_fr=='local')
+    path_keys = ['coef', 'j', 'n', 'n1_max', 'j_fr', 'n_fr', 'spin']
+
+    # Zeroth order
+    U_0 = next(jtfs_gen)
+    assert np.allclose(U_0['coef'], U_0_in)
+    assert U_0['n'] == ()
+    assert U_0['j'] == ()
+
+    # First and second order
+    S_jtfs = list(jtfs_gen)
+
+    # Check that all path keys are unique
+    ns = [path['n'] for path in S_jtfs]
+    assert len(ns) == len(set(ns))
+
+    # Check that order 1 comes before order 2
+    orders = [len(path['n']) for path in S_jtfs]
+    assert set(orders) == {1, 2}
+    assert orders == sorted(orders)
+
+    # Test first order
+    S1_jtfs = filter(lambda path: len(path['n'])==1, S_jtfs)
+    for path in S1_jtfs:
+        # Check path format
+        assert path.keys() == set(path_keys)
+
+        # Check that first-order spins are nonnegative
+        assert path['spin'] >= 0
+
+        # Check frequency padding
+        assert path['n1_max'] < S._N_padded_fr
+
+        # Check that first-order coefficients have the same temporal stride
+        stride = 2**max(S.log2_T - S.oversampling, 0)
+        assert path['coef'].shape[-1] == (S._N_padded // stride)
+
+        # Check that frequential stride works as intended
+        stride_fr = 2**max(path['j_fr'] - S.oversampling_fr, 0)
+        assert (path['coef'].shape[-2]*stride_fr) == S._N_padded_fr
+
+        # Check that padding is sufficient
+        midpoint = (path['n1_max'] + S._N_padded_fr) // (2*stride_fr)
+        avg_value = np.mean(np.abs(path['coef'][midpoint, :]))
+        assert avg_value < 1e-5
+
+    # Test second order
+    S2_jtfs = filter(lambda path: len(path['n'])==2, S_jtfs)
+    for path in S2_jtfs:
+        # Check path format
+        assert path.keys() == set(path_keys)
+
+        # Check frequency padding
+        assert path['n1_max'] < S._N_padded_fr
+
+        # Check that temporal stride works as intended
+        stride = 2**max(path['j'][1] - S.oversampling, 0)
+        assert path['coef'].shape[-1] == (S._N_padded // stride)
+
+        # Check that frequential stride works as intended
+        stride_fr = 2**max(path['j_fr'] - S.oversampling_fr, 0)
+        assert (path['coef'].shape[-2]*stride_fr) == S._N_padded_fr
```

Comment:
This has already been tested from line 224 to 226.

Reply:
224-226 tests the first order. here we're testing the second order.